### ISSUE TITLE
Left/right click on indicator icon broken

### ISSUE THREAD

**Issue by SwooshyCueb** [mate-desktop/mate-indicator-applet#33]
#### Expected behaviour
Left clicking indicator icon performs left click action on indicator icon.
Right clicking indicator icon performs right click action on indicator icon.

#### Actual behaviour
Left clicking indicator icon performs *right click* action on indicator icon.
Right clicking indicator icon opens the context menu for the *indicator applet*,
This breaks programs like flameshot which worked fine in the old gtk2 version of mate-panel/mate-indicator-applet.

#### Steps to reproduce the behaviour
Right or left click on an indicator icon.

#### MATE general version
1.20

#### Package version
mate-indicator-applet 1.20.0-1
mate-panel 1.20.1-3ubuntu1

#### Linux Distribution
Ubuntu MATE 18.04

**Comment by selectiveduplicate** [comment]
Can't reproduce the problem; everything's fine here!

**Comment by cristian64** [comment]
@SwooshyCueb This behaviour change is certainly the most annoying issue I have seen in Ubuntu MATE 18.04.

**Comment by derex** [comment]
Same problem for me on ubuntu 18.04lts with mate: 1.20.1,  indicator-applet: 1.20. Any solutions ?

**Comment by goffyara** [comment]
+1. it's annoying. Terribly unusual.

**Comment by nclm** [comment]
Yes I came here for reporting that issue, this time with QJoyPad, an app that works for the tray area. In this application, left clicking is meant to perform the main action (setting up a game controller), while right clicking allow you to switch in between different presets. The app is unusable on MATE: the right-click menu appears on left click, the applet menu on right click, and the main action is simply not accessible.

**Comment by nclm** [comment]
Yes I came here for reporting that issue (after posting it mistakenly on https://github.com/mate-desktop/mate-panel/issues/976), this time with QJoyPad, an app that only works from the tray area.

In this application, left clicking is meant to perform the main action (setting up a game controller), while right clicking allow you to switch in between different presets.

The app is unusable on MATE: the right-click menu appears on left click, the applet menu on right click, and the main action is simply not accessible.

**Comment by vkareh** [comment]
> #### Actual behaviour
> Left clicking indicator icon performs right click action on indicator icon.
> Right clicking indicator icon opens the context menu for the indicator applet.

This seems like the expected behavior to me. I don't know of any applet that behaves differently than this: right click opens context menu for the applet, left click executes the applet's primary action.

Why is everyone reporting here that this one applet should behave opposite from every other applet? What am I missing?

**Comment by SwooshyCueb** [comment]
> > #### Actual behaviour
> > Left clicking indicator icon performs right click action on indicator icon.
> > Right clicking indicator icon opens the context menu for the indicator applet.
> 
> This seems like the expected behavior to me. I don't know of any applet that behaves differently than this: right click opens context menu for the applet, left click executes the applet's primary action.
> 
> Why is everyone reporting here that this one applet should behave opposite from every other applet? What am I missing?

Because indicator icons have always had both a right-click and left-click action, and it is a breaking change to remove the ability for an indicator icon to use both of these actions.

**Comment by fukawi2** [comment]
Reporting in with the same problem as OP - I can't use Flameshot to take screenshots due to left-click opening Flameshot's right-click menu, and right-click opening indicator app menu.

It works as expected on my Arch desktop with 1.22.1, but on Ubuntu 18.04.3 with 1.20.1 it's broken - so not sure if this has been fixed, or if this is perhaps an Ubuntu Mate bug?

**Comment by zenden2k** [comment]
Have the same issue on Ubuntu 18.04.3 and Mate 1.20.1
Upgrading to Mate 1.22.1 did not solve this problem.

**Comment by chexmo** [comment]
Same happens in Ubuntu MATE 19.10
This is one of the little few things i don't like about MATE right now.

**Comment by slcpunk** [comment]
I also wish there was a workaround for this.  If someone reading doesn't understand, here's another example.  BirdTray is an email indicator.  The expected behavior ( like the items in the dock ) is that a left click will raise/lower (hide/show) the Thunderbird window.  The right click would bring up the context menu.  As others describe, instead the left click brings up the context menu and a right click brings up the App Indicator menu instead.  I realize why - otherwise it would be hard to see the app indicator menu - but I think the solution is to have the click work that way on the "handle" of the add indicator applet.   Note that Mate Dock handles this correctly ( and even has an option to change it ) passing through the left clicks properly.
